win32/Lib/pywin32_bootstrap.py

```diff
@@ -7,22 +7,27 @@
 # If Python has `os.add_dll_directory()`, we need to call it with this path.
 # Otherwise, we add this path to PATH.
 import os
-import site
 
-# The directory should be installed under site-packages.
 
-dirname = os.path.dirname
-# This is to get the "...\Lib\site-packages" directory
-# out of this file name: "...\Lib\site-packages\win32\Lib\pywin32_bootstrap.py".
-# It needs to be searched when installed in virtual environments.
-level3_up_dir = dirname(dirname(dirname(__file__)))
+try:
+    Exc = ModuleNotFoundError  # Introduced in Python 3.6
+except NameError:
+    Exc = ImportError
 
-site_packages_dirs = getattr(site, "getsitepackages", lambda: [])()
-if level3_up_dir not in site_packages_dirs:
-    site_packages_dirs.insert(0, level3_up_dir)
+try:
+    import pywin32_system32
+except Exc:
+    path_iterator = iter([])
+else:
+    # We're guaranteed only that __path__: Iterable[str]
+    # https://docs.python.org/3/reference/import.html#__path__
+    path_iterator = iter(pywin32_system32.__path__)
 
-for site_packages_dir in site_packages_dirs:
-    pywin32_system32 = os.path.join(site_packages_dir, "pywin32_system32")
+try:
```

Comment:
We can probably simplify this by putting the rest of the body under this else - ie, just have a `pass` on the import error and everything else in the `else`?

Reply:
I _think_ df3e334  is responsive to this.